protocol/src/common/protocol.ts

```diff
@@ -350,6 +350,37 @@ export interface WorkspaceClientCapabilities {
 	codeLens?: CodeLensWorkspaceClientCapabilities;
 }
 
+export interface FileClientCapabilities {
```

Comment:
Now that I think about the name we should may be call this FileManipulation to make it clear that these are not file events in the classical sense of file watching. We should also add a sentence to make this clear.

Reply:
You used `FileOperations` further down which is a good name.

Reply:
Done!